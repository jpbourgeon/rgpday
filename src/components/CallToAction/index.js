import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import { Link as RouterLink } from '@reach/router'
import Link from '@material-ui/core/Link'
import Typography from '@material-ui/core/Typography'
import Grid from '@material-ui/core/Grid'
import Paper from '@material-ui/core/Paper'
import Hidden from '@material-ui/core/Hidden'

const styles = theme => ({
  cta: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 6,
    textAlign: 'center'
  },
  link: {
    '&:hover': {
      cursor: 'pointer',
      textDecoration: 'underline'
    }
  }
})

const LinkToContact = props => <RouterLink to='/contact' {...props} />

const CallToAction = (props) => {
  const { classes, xs, sm, md } = props
  const cta = (
    <React.Fragment>
      <Link component={LinkToContact} color='secondary'>Contactez-moi&nbsp;!</Link><br />
    pour organiser le RGPDay dans votre Ã©tablissement
    </React.Fragment>
  )
  return (
    <Grid container spacing={40} alignItems='center' direction='column'>
      <Grid item xs={xs} sm={sm} md={md}>
        <Paper className={classes.cta} elevation={1}>
          <Hidden mdUp><Typography variant='h6'>{cta}</Typography></Hidden>
          <Hidden smDown><Typography variant='h6'>{cta}</Typography></Hidden>
        </Paper>
      </Grid>
    </Grid>
  )
}

CallToAction.propTypes = {
  classes: PropTypes.object.isRequired,
  xs: PropTypes.number,
  sm: PropTypes.number,
  md: PropTypes.number
}

CallToAction.defaultProps = {
  xs: 12,
  sm: 12,
  md: 6
}

export default withStyles(styles)(CallToAction)
