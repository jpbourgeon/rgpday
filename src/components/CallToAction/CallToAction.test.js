/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import CallToAction from './'
import jssSerializer from 'src/utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The CallToAction component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<CallToAction />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
