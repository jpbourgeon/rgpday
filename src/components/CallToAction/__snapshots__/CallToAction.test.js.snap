// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CallToAction component should render correctly 1`] = `
<div
  className="container-3 40-29 column-6 center-11"
>
  <div
    className="item-4 12-43 6-65"
  >
    <div
      className="root-100 elevation1-103 rounded-101 cta-1"
    >
      <h6
        className="root-127 h6-144"
      >
        Si ce format vous intéresse, 
        <a
          className="root-127 colorSecondary-158 inline-162 root-163 underlineHover-165 link-2"
          href="https://fr.gravatar.com/jpbourgeon"
          rel="noopener"
          target="_blank"
        >
          contactez-moi
        </a>
      </h6>
      <p
        className="root-127 body1-136"
      >
        pour organiser le RGPDay dans votre établissement !
      </p>
    </div>
  </div>
</div>
`;
